### PR TITLE
Small refactoring of SessionItem

### PR DESCRIPTION
**Pull request by waltervh**

### CODE REVIEW

File: GUI/coregui/Models/SessionItemUtils.cpp [hunk 1]
```diff
@@ -25,6 +25,11 @@ int SessionItemUtils::ParentRow(const SessionItem& item)
     return -1;
 }
 
+bool SessionItemUtils::HasCHildren(const SessionItem& item)
```
Comment (gpospelov) [inline GUI/coregui/Models/SessionItemUtils.cpp]: I would not move hasChildren to utils for the sake of similarity of our SessionItem and QStandardItem. QStandardItem has a method hasChildren

File: GUI/coregui/Views/InstrumentWidgets/BeamEditorWidget.cpp [hunk 2]
```diff
@@ -59,12 +59,12 @@ BeamEditorWidget::BeamEditorWidget(QWidget* parent)
     mainLayout->addStretch();
     setLayout(mainLayout);
 
-    connect(m_wavelengthPresenter, SIGNAL(onDialogRequest(SessionItem*, QString)),
```
Comment (gpospelov) [inline GUI/coregui/Views/InstrumentWidgets/BeamEditorWidget.cpp]: While fully supporting an idea of using new signal/slot syntax, I would switch to it only while making some other (bigger) refactoring in a widget.

File: GUI/coregui/Models/SessionItemUtils.cpp (hunk 1 above)
Comment (waltervh) [reply GUI/coregui/Models/SessionItemUtils.cpp]: Right, I will move it back

File: GUI/coregui/Views/InstrumentWidgets/BeamEditorWidget.cpp (hunk 2 above)
Comment (waltervh) [reply GUI/coregui/Views/InstrumentWidgets/BeamEditorWidget.cpp]: I did it here to remove the Q_DECLARE_METATYPE with its obscure comment

Comment (gpospelov) [reply GUI/coregui/Views/InstrumentWidgets/BeamEditorWidget.cpp]: ah, ok

**Pull request merged**